homeassistant/components/volvo/binary_sensor.py

```diff
@@ -0,0 +1,416 @@
+    ),
+    # tyres endpoint
+    VolvoCarsTireDescription(
+        key="tire_front_right",
+        api_field="frontRight",
+    ),
+    # tyres endpoint
+    VolvoCarsTireDescription(
+        key="tire_rear_left",
+        api_field="rearLeft",
+    ),
+    # tyres endpoint
+    VolvoCarsTireDescription(
+        key="tire_rear_right",
+        api_field="rearRight",
+    ),
+    # windows endpoint
+    VolvoCarsWindowDescription(
+        key="window_front_left",
+        api_field="frontLeftWindow",
+    ),
+    VolvoCarsWindowDescription(
+        key="window_front_right",
+        api_field="frontRightWindow",
+    ),
+    VolvoCarsWindowDescription(
+        key="window_rear_left",
+        api_field="rearLeftWindow",
+    ),
+    VolvoCarsWindowDescription(
+        key="window_rear_right",
+        api_field="rearRightWindow",
+    ),
+    VolvoCarsWindowDescription(
+        key="sunroof",
+        api_field="sunroof",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: VolvoConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up binary sensors."""
+    coordinators = entry.runtime_data
+    entities = [
+        VolvoBinarySensor(coordinator, description)
+        for coordinator in coordinators
+        for description in _DESCRIPTIONS
+        if description.api_field in coordinator.data
+    ]
+
+    async_add_entities(entities)
+
+
+class VolvoBinarySensor(VolvoEntity, BinarySensorEntity):
+    """Volvo binary sensor."""
+
+    entity_description: VolvoBinarySensorDescription
+
+    def __init__(
+        self,
+        coordinator: VolvoBaseCoordinator,
+        description: VolvoBinarySensorDescription,
+    ) -> None:
+        """Initialize entity."""
+        self._attr_extra_state_attributes = {}
+
+        super().__init__(coordinator, description)
+
+    def _update_state(self, api_field: VolvoCarsApiBaseModel | None) -> None:
+        """Update the state of the entity."""
+        if api_field is None:
+            self._attr_is_on = None
+            return
+
+        assert isinstance(api_field, VolvoCarsValue)
+
+        self._attr_is_on = (
+            api_field.value in self.entity_description.on_values
+            if isinstance(api_field.value, str)
+            and api_field.value.upper() != API_NONE_VALUE
+            else None
+        )
+
+        if self.entity_description.api_value_in_attributes:
+            attribute_value = (
+                api_field.value
+                if self.entity_description.api_value_attribute_fn is None
+                else self.entity_description.api_value_attribute_fn(api_field.value)
+            )
+
+            if isinstance(attribute_value, str):
+                attribute_value = value_to_translation_key(attribute_value)
+
+            self._attr_extra_state_attributes[
+                self.entity_description.api_value_attribute_name
+            ] = attribute_value
```

Comment:
Generally we don't like extra state attributes, can you add what kind of attributes we add here? (I tried to understand it, but it's probably quicker to ask)

Reply:
Looking at the strings.json I see oil level, tire pressure, etc, I think they can be their own enum sensor and maybe disabled by default

Reply:
It seems we didn't remove the attributes?